src/components/MAPI/apps/ClusterDetailIngress.tsx

```diff
@@ -123,6 +129,18 @@ const ClusterDetailIngress: React.FC<
     return Boolean(app);
   }, [appList?.items]);
 
+  const k8sEndpoint = useMemo(() => {
+    if (rest.k8sEndpoint) {
+      return rest.k8sEndpoint;
+    }
+
+    return cluster
+      ? isCAPGCluster(cluster) || isCAPACluster(cluster)
+        ? getClusterK8sAPIUrl(cluster, provider)
+        : capiv1beta1.getClusterK8sAPIUrl(cluster)
+      : undefined;
```

Comment:
Instead of having 2 similar functions both named `getClusterK8sAPIUrl`, would it be possible to combine them and move the provider checks there?

Reply:
After refactoring:
`getClusterK8sAPIUrl ` was moved to the `MAPI/utils.ts` and is used to display KUBERNETES API on a cluster Overview page.
`getClusterK8sEndpoint` is now used for displaying instructions on a cluster Ingress tab and was moved to the `MAPI/apps/utils.ts`.